src/snowflake/cli/plugins/connection/commands.py

```diff
@@ -73,9 +73,10 @@ def callback(value: str):
     return callback
 
 
-def _password_callback(value: str):
-    if value:
+def _password_callback(ctx: Context, value: str):
+    if value and not ctx.default_map:
```

Comment:
This checks whether `default_map` is not None / is not empty. Is this intended?

```py
In [1]: 'a' if {} else 'b'
Out[1]: 'b'
```

```suggestion
    if value and not ctx.default_map.get("password_callback_shown", False):
```

Reply:
Yes. ctx.default_map is an Optional[Dict] - it`s not present in the first run, so this passes.
If callback i called second time, the dicts is initialised, and the if condition fails.